runtime/Java/src/org/antlr/v4/runtime/ParserRuleContext.java

```diff
@@ -272,7 +272,14 @@ public TerminalNode getToken(int ttype, int i) {
 
 	@Override
 	public Interval getSourceInterval() {
-		if ( start==null || stop==null ) return Interval.INVALID;
+		if ( stop==null ) {
+			if ( start!=null ) {
+				return Interval.of(start.getTokenIndex(), start.getTokenIndex()-1); // empty
+			}
+			else {
+				return Interval.of(0, -1);
+			}
+		}
 		return Interval.of(start.getTokenIndex(), stop.getTokenIndex());
```

Comment:
:bulb: One would expect the total number of tokens from the input sequence contained in the source interval (including off-channel tokens) to be the following:

```java
Interval sourceInterval = getSourceInterval();
int tokenCount = sourceInterval.b - sourceInterval.a + 1;
```

However, in the specific case where off-channel tokens appear before an empty parse tree, `tokenCount` could be negative. This could be addressed by changing the `stop==null` condition above to `stop==null || stop.getTokenIndex()<start.getTokenIndex()`.